### ISSUE TITLE
Adding NVIDIA-SMI like information

### ISSUE THREAD

**Issue by groadabike** [PyTorchLightning/pytorch-lightning#2074]
## 🚀 Feature

- Add the GPU usage information during training.

### Motivation

Most of the research is done on HPC. Therefore, if I want to see the GPU RAM and usage of my job, I have to open a secondary screen to run "watch nvidia-smi" or "nvidia-smi dmon".
Have this info saved in the logs will help to:
1. See if I have space for larger batches
2. Report the correct resources needed to replicate my experiment.

### Pitch

When training starts, report the GPU RAM and the GPU usage together with loss and v_num
  
### Alternatives

After the first epoch is loaded into the GPU, log the GPU RAM and the GPU usage

**Comment by Borda** [comment]
add 1) you can use Batch finder
add 2) how it is different from a logger?
cc: @jeremyjordan @SkafteNicki

**Comment by SkafteNicki** [comment]
If your goal is just to optimize for batch size, then the batch finder may be what you are looking for.
If we where to log the resource usage, I guess that we could write a callback similar to `LearningRateLogger` that extracts this information (through `nvidia-smi` or maybe `gpustat`?) and logs these numbers to the `logger` of the trainer.

**Comment by groadabike** [comment]
Hi @SkafteNicki ,
In fact, what I am looking are both things. First, optimize the batch size that fits in my GPU and then to keep logging the GPU usage.
I will try to use https://github.com/sicara/gpumonitor that implements a pl's callback.

**Comment by Borda** [comment]
@groadabike mind send a PR with PL callback?

**Comment by groadabike** [comment]
Hi @Borda , I try to use the gpumonitor callback but it didn't work in my HPC.
For some reason, the training stop waiting for something. 
I can't send a PR as I don't have any callback implemented.
I am still in need to know the GPU utilisation because I know I have a bottleneck in the dataloader (using the profiler), but I don't know for how long the GPU is waiting for the next batch.
Will back to you when I solve this issue

**Comment by groadabike** [comment]
Hi Borda,
I have a first attempt for my Callback to measure the GPU usage and GPU "dead" periods.
Can you take a look at it and give me your feedback?
I am doing several measures and logging in tensorboard:

1- Time between batches  - the time between the end of one batch and the start of the next.
2- Time in batch - the time between the start and end of one batch
![Screenshot from 2020-08-11 12-51-21](https://user-images.githubusercontent.com/17277802/89894171-930d6d80-dbd1-11ea-8acf-3d6a9ca708d3.png)


3- GPU utilisation - % of GPU utilisation measured at the beginning and end of each batch. 
![Screenshot from 2020-08-11 12-52-17](https://user-images.githubusercontent.com/17277802/89894154-89840580-dbd1-11ea-9cf3-453df2bd6fb1.png)

4- GPU memory used
![Screenshot from 2020-08-11 12-53-46](https://user-images.githubusercontent.com/17277802/89894276-bdf7c180-dbd1-11ea-9fb4-f1d3cf0d3941.png)

5- GPU memory free
![Screenshot from 2020-08-11 12-53-32](https://user-images.githubusercontent.com/17277802/89894282-c0f2b200-dbd1-11ea-8646-793ef60ed4a3.png)


[gpuusage_callback.zip](https://github.com/PyTorchLightning/pytorch-lightning/files/5056724/gpuusage_callback.zip)

**Comment by SkafteNicki** [comment]
@groadabike i think that looks like a great addition. I you want to submit a PR, feel free :)
Personally I would also add flags for temperature (`temperature.gpu` and `temperature.memory` query) and fans (`fan.speed` query), both can be disabled by default. For the `memory_utilization` flag I would also log `utilization.memory`.

**Comment by SkafteNicki** [comment]
closing this as it was solved by PR https://github.com/PyTorchLightning/pytorch-lightning/pull/2932
